spectrum/xpub.py

```diff
@@ -42,6 +48,24 @@ def dashboard(self):
         dashboard_button.click()
         return XpubDashboardPage(self._driver)
 
+    def _dismiss_cookie_notice(self):
+        try:
+            cookie_notice_button = self._driver.find_element_by_css_selector(self.CSS_COOKIE_NOTICE_BUTTON)
+            _info_log("Found cookie notice button %s `%s`", self.CSS_COOKIE_NOTICE_BUTTON, cookie_notice_button.text)
+        except NoSuchElementException:
+            _info_log("Found no cookie notice button %s", self.CSS_COOKIE_NOTICE_BUTTON)
+            return
+
+        cookie_notice_button.click()
+        _info_log("Clicked cookie notice button %s", self.CSS_COOKIE_NOTICE_BUTTON)
+        WebDriverWait(self._driver, self.TIMEOUT_COOKIE_NOTICE_CLOSING).until(lambda driver: self._cookie_notice_closed())
+
+    def _cookie_notice_closed(self):
+        LOGGER.debug("Cookie notice visibility check")
```

Comment:
This waiting condition loses 10 seconds per test because it is a negative check. When we are testing for the absence, the Selenium implicit wait we set (10 seconds) will wait to see if the element is truly absent or it has just not appeared yet.

If there was something we can _positively_ test for in the DOM, this will be faster as we can move on as soon as "element that denotes the cookie notice has been dismissed" appears.

Reply:
Since we are on topic, the similarly fragile https://github.com/elifesciences/elife-spectrum/blob/master/spectrum/xpub.py#L190-L202 counts `h2` elements to understand when the `PeoplePicker` has been closed.

Reply:
Thanks this is useful to know. I've raised https://github.com/elifesciences/elife-xpub/issues/1710